073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,339 @@
+
+#### Fig 2: Clients CA certificate and component relationships
+
+![Clients CA relationships](./images/073-clients-ca.svg)
+
+### Existing Kubernetes Secrets
+
+Strimzi currently uses the following Kubernetes Secrets to store the CA certificates and private keys:
+* <CLUSTER_NAME>-cluster-ca-cert for the Cluster CA root certificate
+* <CLUSTER_NAME>-cluster-ca for the Cluster CA private key
+* <CLUSTER_NAME>-clients-ca-cert for the Clients CA root certificate
+* <CLUSTER_NAME>-clients-ca for the Clients CA private key
+
+Strimzi currently uses the following Kubernetes Secrets to store server certificates (signed by the cluster CA) that components will present to clients:
+* <CLUSTER_NAME>-kafka-brokers for Kafka brokers
+* <CLUSTER_NAME>-zookeeper-nodes for ZooKeeper nodes
+
+Strimzi currently uses the following Kubernetes Secrets to store client certificates (signed by the cluster CA) that components will present to Kafka or ZooKeeper when making requests:
+* <CLUSTER_NAME>-cluster-operator-certs for the Cluster Operator
+* <CLUSTER_NAME>-entity-topic-operator-certs for the Entity Topic Operator
+* <CLUSTER_NAME>-entity-user-operator-certs for the Entity User Operator
+* <CLUSTER_NAME>-kafka-exporter-certs for the Kafka Exporter
+* <CLUSTER_NAME>-cruise-control-certs for Cruise Control
+
+### Existing Kubernetes Secret and Pod Annotations
+
+Strimzi uses both the notion of a "generation" and hashes when keeping track of certificates.
+
+A generation is associated with a specific certificate or private key for either the Cluster or Clients CA.
+If Strimzi is managing the CA, it increments the generation each time the certificate or private key is renewed or replaced.
+If the user is managing the CA, they are responsible for incrementing the generation.
+The generations are then used during the reconcile loop to track the state of trust (as described in the example flows below).
+
+Strimzi uses the following annotations on Kubernetes Secrets:
+* strimzi.io/ca-cert-generation to indicate the generation of the Cluster CA or Client CA certificate in the Kubernetes Secret
+* strimzi.io/ca-key-generation to indicate the generation of the Cluster CA or Client CA private key in the Kubernetes Secret
+
+Strimzi uses the following annotations on Kubernetes Pods and Kubernetes Secrets to indicate the generation of the Cluster CA certificate used to sign the certificates that are either stored in the Kubernetes Secret, or presented by the Kubernetes Pod:
+* strimzi.io/cluster-ca-cert-generation for certificates signed by the Cluster CA
+* strimzi.io/clients-ca-cert-generation for certificates signed by the Clients CA
+
+Strimzi uses the following annotation on Kubernetes Pods to indicate the generation of the Cluster CA private key used to sign the cluster CA certificate that the pod currently trusts:
+* strimzi.io/cluster-ca-key-generation
```

Comment:
I would specificy "components" here, so something like "Kubernetes Pods components" even because technically it's not the pod itself trusting a certificate but the component/application running within it.